llvm/lib/Transforms/Instrumentation/AddressSanitizer.cpp

```diff
@@ -3505,7 +3505,7 @@ void FunctionStackPoisoner::processStaticAllocas() {
       SplitBlockAndInsertIfThenElse(Cmp, Ret, &ThenTerm, &ElseTerm);
 
       IRBuilder<> IRBPoison(ThenTerm);
-      if (StackMallocIdx <= 4) {
+      if (ASan.MaxInlinePoisoningSize != 0 && StackMallocIdx <= 4) {
```

Comment:
Should it be rather 
```
  int ClassSize = kMinStackMallocSize << StackMallocIdx;
  if (ASan.MaxInlinePoisoningSize < ClassSize / L.Granularity && StackMallocIdx <= 4) {
        ShadowAfterReturn.resize(ClassSize / L.Granularity,
                                 kAsanStackUseAfterReturnMagic);
```

Reply:
Not sure if I understand - minimum ClassSize is going to be 64 (kMinStackMallocSize << StackMallocIdx), L.Granularity = 8 , so if ASan.MaxInlinePoisoningSize == 0, the whole condition will be true, so inlining will happen. ASan.MaxInlinePoisoningSize == 0 should guarantee not inlining. So I'd rather keep it explicitly the way I had it, unless I misunderstood your suggestion?

Reply:
My comment is about the case when `MaxInlinePoisoningSize > 0`
There is no point in  `copyToShadow` the frame is larger then `MaxInlinePoisoningSize`

Reply:
Perhaps I used wrong setting for this (MaxInlinePoisoningSize), which caused misunderstanding.  I want no inline shadow accesses at all. So this optimization `StackMallocIdx <= 4)` for small frames should be turned off (use code under else)  when  ClInstrumentationWithCallsThreshold() == 0. Would that work / be more explicit when I change it to be this way?

Reply:
Or, ASAN.InstrumentationWithCallsThreshold as it is available in this code today

Reply:
> Or, ASAN.InstrumentationWithCallsThreshold as it is available in this code today

`InstrumentationWithCallsThreshold` is for checks
`MaxInlinePoisoningSize` for poisoning, like the code in question